src/main/java/com/telefonica/iot/cygnus/sinks/OrionSTHSink.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2015 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-cygnus (FI-WARE project).
+ *
+ * fiware-cygnus is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-cygnus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-cygnus. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with iot_support at tid dot es
+ */
+package com.telefonica.iot.cygnus.sinks;
+
+import com.telefonica.iot.cygnus.containers.NotifyContextRequest;
+import static com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink.LOGGER;
+import com.telefonica.iot.cygnus.utils.Constants;
+import com.telefonica.iot.cygnus.utils.Utils;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ *
+ * @author frb
+ */
+public class OrionSTHSink extends OrionMongoBaseSink {
+
+    /**
+     * Constructor.
+     */
+    public OrionSTHSink() {
+        super();
+    } // OrionSTHSink
+    
+    @Override
+    void persist(Map<String, String> eventHeaders, NotifyContextRequest notification) throws Exception {
+        // get some header values
+        Long recvTimeTs = new Long(eventHeaders.get("timestamp"));
+        String fiwareService = eventHeaders.get(Constants.HEADER_SERVICE);
+        String fiwareServicePath = eventHeaders.get(Constants.HEADER_SERVICE_PATH);
+        String[] destinations = eventHeaders.get(Constants.DESTINATION).split(",");
```

Comment:
If no attributes are notified, the code protect itself by checking if the attribute list is null or empty. A warning is raised if such a thing occurs.

If no attribute is aggregatable, a warning is raised as well. Regarding this, I made a comment on the STH counterpart about the convenience of raising an error or not... as you can see, I think a warning is enough :)

Nevertheless, I agree there is a problem regarding the header inspection when no attributes are notified, since this header will contain a null value, or in the best case, a empty list of destinations. I will fix it, thanks!